### PR TITLE
[Stable Diffusion] Add components function

### PR DESCRIPTION
**Pull request by patrickvonplaten**

### CODE REVIEW

File: src/diffusers/pipeline_utils.py [hunk 1]
```diff
@@ -541,6 +541,41 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
...
+        expected_modules = set(inspect.signature(self.__init__).parameters.keys()) - set(["self"])
+
+        if set(components.keys()) != expected_modules:
+            raise ValueError(
+                f"{self} has been incorrectly initialized or {self.__class__} is incorrectly implemented. Expected"
+                f" {expected_modules} to be defined, but {components} are defined."
+            )
+
+        return components
+
```
Comment (patil-suraj) [inline src/diffusers/pipeline_utils.py:578]: Very cool!

File: docs/source/api/pipelines/stable_diffusion.mdx [hunk 2]
```diff
@@ -17,6 +17,26 @@ For more details about how Stable Diffusion works and how it differs from the ba
...
+...     StableDiffusionPipeline,
+...     StableDiffusionImg2ImgPipeline,
+...     StableDiffusionInpaintPipeline,
+... )
+
+>>> img2text = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")
+>>> img2img = StableDiffusionImg2ImgPipeline(**img2text.components)
+>>> inpaint = StableDiffusionInpaintPipeline(**img2text.components)
+
+>>> # now you can use img2text(...), img2img(...), inpaint(...) just like the call methods of each respective pipeline
```
Comment (patil-suraj) [inline docs/source/api/pipelines/stable_diffusion.mdx:37]: Nice example! We should also explain a bit more details, for example, the `revision`, `dtype`.
Also we should mention that each pipeline needs to be placed on a `device` so it has it's `self.device` set.

**Review by patil-suraj — APPROVED** [review]
Looks very cool, just left one comment.

File: docs/source/api/pipelines/stable_diffusion.mdx (hunk 2 above)
Comment (patrickvonplaten) [reply docs/source/api/pipelines/stable_diffusion.mdx:37]: Yeah leaving this for a follow-up PR ;-)

**Pull request merged**